pkg/operator/apis/monitoring/v1/common_types.go

```diff
@@ -181,3 +181,17 @@ var invalidLabelCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
 func sanitizeLabelName(name string) prommodel.LabelName {
 	return prommodel.LabelName(invalidLabelCharRE.ReplaceAllString(name, "_"))
 }
+
+// validateDistinctJobNames returns error if any job name duplicates.
+// This would also result in an error on Prometheus server config parsing.
+func validateDistinctJobNames(jobs []*promconfig.ScrapeConfig) error {
+	jobToIndex := map[string]int{}
+	for i, job := range jobs {
+		if dup, ok := jobToIndex[job.JobName]; ok {
+			// TODO(bwplotka): Consider adding JobSuffix field or so for https://github.com/GoogleCloudPlatform/prometheus-engine/issues/479
+			return fmt.Errorf("duplicated job name %v for endpoints with index %d and %d. Consider creating a separate resource for each job", job.JobName, dup, i)
```

Comment:
nit(non-blocking): For users, a job name is abstracted away from them. We construct them rather arbitrarily. For example, why there's no reason why we couldn't encode scrape intervals into them to help make them unique and further abstract them away. As such, I believe this error message may be confusing for them.

What the user actually has to do is create a resource with either a different namespace, name or port depending on the resource type. I wonder if we could spell something like that out here?

Reply:
I agree I can give more constructive hints, but I disagree job name is abstracted away. While we generate, validate and sanitize it, users definitely has to be aware of what job means and how is constructed when using those metrics via PromQL/other means. I think it should be fine to mention the job abstraction here.

Essentially they might want to aggregate by job label when alering, checking it when querying, retrieving series to tell what *Monitoring CR configured it (in some way, what application we are monitoring).

Let me improve the message..